internal/store/store.go

```diff
@@ -52,12 +54,32 @@ type Store struct {
 	client        interface{}
 }
 
-func newTransport(c config.Config) *digest.Transport {
-	t := &digest.Transport{
-		Username: c.PublicAPIKey(),
-		Password: c.PrivateAPIKey(),
+func customCATransport(ca []byte) http.RoundTripper {
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(ca)
+	tlsClientConfig := &tls.Config{
+		InsecureSkipVerify: false,
+		RootCAs:            caCertPool,
+	}
+	return &http.Transport{
+		ResponseHeaderTimeout: responseHeaderTimeout,
+		TLSHandshakeTimeout:   tlsHandshakeTimeout,
+		DialContext: (&net.Dialer{
+			Timeout:   timeout,
+			KeepAlive: keepAlive,
+		}).DialContext,
+		MaxIdleConns:          maxIdleConns,
+		MaxIdleConnsPerHost:   maxIdleConnsPerHost,
+		Proxy:                 http.ProxyFromEnvironment,
+		IdleConnTimeout:       idleConnTimeout,
+		ExpectContinueTimeout: expectContinueTimeout,
+		TLSClientConfig:       tlsClientConfig,
```

Comment:
TL;DR: you can only set this on construction and the client was relying on reflection to "re set" this if needed, that reflection was failing when using the `digest.Transport` as the lib is agnostic about the authentication we use